### PR TITLE
Fix crash when meet unknown column type from TiDB

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/7943

Problem Summary:

### What is changed and how it works?

* Check whether the type exist before calling the lambda in `TypeMapping::getDataType` to avoid crash
* Move `KVStore::handleApplySnapshot` to `RegionBench::handleApplySnapshot`
* Add logging about the bytes_on_disk when pre-handling snapshot
* Skip when `BlobStore::gc` meets total_page_size == 0 after https://github.com/pingcap/tiflash/pull/7719
* Add some unit tests for handling apply snapshot in KVStore

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by JaySon-Huang** [comment]
@CalvinNeo @lidezhu PTAL

**Comment by lidezhu** [comment]
Just be curious, why there will be unknown column type?

File: dbms/src/Debug/dbgFuncMockRaftSnapshot.cpp [hunk 1]
```diff
@@ -245,7 +245,8 @@ void MockRaftCommand::dbgFuncRegionSnapshot(Context & context, const ASTs & args
     auto end_decoded_key = RecordKVFormat::decodeTiKVKey(end_key);
 
     // Mock to apply an empty snapshot for region[region-id]
-    tmt.getKVStore()->handleApplySnapshot(
+    RegionBench::handleApplySnapshot(
```
Comment (lidezhu) [inline dbms/src/Debug/dbgFuncMockRaftSnapshot.cpp:248]: why move it to `RegionBench`?

**Comment by JaySon-Huang** [comment]
> Just be curious, why there will be unknown column type?

I'm not sure why there are unknown column type. I've try to parse the JSON schema that get from the issue logging, but all goes well.
So I just add a check to ensure what happen when it happen next time. Hopefully we will get an exception with more message and no crash after this.

File: dbms/src/Debug/dbgFuncMockRaftSnapshot.cpp (hunk 1 above)
Comment (JaySon-Huang) [reply dbms/src/Debug/dbgFuncMockRaftSnapshot.cpp:248]: This is just a debug function and what this function call (`preHandleSnapshotToFiles`/`applyPreHandledSnapshot`) are public methods. So I want to move this debugging function out of KVStore to reduce the code complexity on KVStore

**Review by lidezhu — APPROVED** [review]
LGTM

**Review by CalvinNeo — APPROVED** [review]
lgtm

**Pull request merged**